src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -67,33 +67,29 @@ public static class Reflect
         /// <returns>The array of methods found</returns>
         public static MethodInfo[] GetMethodsWithAttribute(Type fixtureType, Type attributeType, bool inherit)
         {
-            List<MethodInfo> list = new List<MethodInfo>();
-
             var flags = AllMembers | (inherit ? BindingFlags.FlattenHierarchy : BindingFlags.DeclaredOnly);
-            foreach (MethodInfo method in fixtureType.GetMethods(flags))
-            {
-                if (method.IsDefined(attributeType, inherit))
-                    list.Add(method);
-            }
 
-            list.Sort(new BaseTypesFirstComparer());
```

Comment:
There are half a dozen other instances of `List<>.Sort` that I didn't touch that are suspect as well. We should not be using `List<>.Sort` or `Array.Sort<>` if there is a chance that two different items could compare as equal.